language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -361,24 +381,43 @@ impl CompiledPackage {
             .filter(|unit| matches!(unit.unit, CompiledUnit::Script(_)))
     }
 
+    fn can_load_cached(
+        package: &OnDiskCompiledPackage,
+        resolution_graph: &ResolvedGraph,
+        resolved_package: &ResolvedPackage,
+        is_root_package: bool,
+    ) -> bool {
+        !(package.has_source_changed_since_last_compile(resolved_package)
```

Comment:
This seems to be a complicated logic here. Any thoughts on how we should test this piece?

Reply:
This is going to be rather difficult to write a test for, although I can think of some ways but they're all fairly involved. Would you be OK with landing this PR as-is and then I can put up a separate PR with caching tests?

Reply:
Sounds good!